portal-backend/depmap/private/views.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+from depmap.utilities.sign_bucket_url import sign_url
+from flask import (
+    Blueprint,
+    abort,
+    current_app,
+    redirect,
+)
+from oauth2client.service_account import ServiceAccountCredentials
+
+log = logging.getLogger(__name__)
+
+blueprint = Blueprint(
+    "private", __name__, url_prefix="/private", static_folder="../static"
+)
+
+
+@blueprint.route("/<bucket_name>/<filename>")
```

Comment:
To support directories within the bucket (more specifically, to allow "/" in filename), I think this should be something like 

```
@blueprint.route("/<bucket_name>/<path:filename>")
```

Reply:
Good point! The idea of creating directories within the bucket didn't cross my mind.